File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/winforms/samples/data/comboboxbinding/cs/ComboBoxBinding.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------


namespace Microsoft.Samples.WinForms.Cs.ComboBoxBinding {

    using System;
    using System.ComponentModel;
    using System.Drawing;
    using System.Windows.Forms;
    using System.Data;
    using System.Data.SqlClient;

    public class ComboBoxBinding : System.Windows.Forms.Form {
        private System.ComponentModel.Container components;
        private Microsoft.Samples.WinForms.Cs.ComboBoxBinding.Data.CustomersDataSet customersDataSet1;
        private System.Windows.Forms.TextBox textBoxZip;
        private System.Windows.Forms.ComboBox comboBoxState;
        private System.Windows.Forms.TextBox textBoxCity;
        private System.Windows.Forms.Label labelState;
        private System.Windows.Forms.Label labelZip;
        private System.Windows.Forms.Label labelCity;
        private System.Windows.Forms.TextBox textBoxPosition;
        private System.Windows.Forms.Button buttonMoveFirst;
        private System.Windows.Forms.Button buttonMovePrev;
        private System.Windows.Forms.Button buttonMoveNext;
        private System.Windows.Forms.Button buttonMoveLast;
        private System.Windows.Forms.TextBox textBoxCompany;
        private System.Windows.Forms.Label labelCompanyName;
        private System.Windows.Forms.TextBox textBoxAddress;
        private System.Windows.Forms.TextBox textBoxTitle;
        private System.Windows.Forms.TextBox textBoxContact;
        private System.Windows.Forms.TextBox textBoxID;
        private System.Windows.Forms.Label labelAddress;
        private System.Windows.Forms.Label labelContactTitle;
        private System.Windows.Forms.Label labelContact;
        private System.Windows.Forms.Label labelID;
        private System.Windows.Forms.Panel panelVCRControl;

        //Define the array of states for the drop-down list
        public State[] States  = new State[] {
            new State("Alabama","AL")
           ,new State("Alaska","AK")
           ,new State("Arizona" ,"AZ")
           ,new State("Arkansas","AR")
           ,new State("California" ,"CA")
           ,new State("Colorado","CO")
           ,new State("Connecticut","CT")
           ,new State("Delaware","DE")
           ,new State("District of Columbia","DC")
           ,new State("Florida" ,"FL")
           ,new State("Georgia" ,"GA")
           ,new State("Hawaii"  ,"HI")
           ,new State("Idaho","ID")
           ,new State("Illinois","IL")
           ,new State("Indiana" ,"IN")
           ,new State("Iowa","IA")
           ,new State("Kansas"  ,"KS")
           ,new State("Kentucky","KY")
           ,new State("Louisiana"  ,"LA")
           ,new State("Maine","ME")
           ,new State("Maryland","MD")
           ,new State("Massachusetts","MA")
           ,new State("Michigan","MI")
           ,new State("Minnesota"  ,"MN")
           ,new State("Mississippi","MS")
           ,new State("Missouri","MO")
           ,new State("Montana" ,"MT")
           ,new State("Nebraska","NE")
           ,new State("Nevada"  ,"NV")
           ,new State("New Hampshire","NH")
           ,new State("New Jersey" ,"NJ")
           ,new State("New Mexico" ,"NM")
           ,new State("New York","NY")
           ,new State("North Carolina","NC")
           ,new State("North Dakota" ,"ND")
           ,new State("Ohio","OH")
           ,new State("Oklahoma","OK")
           ,new State("Oregon"  ,"OR")
           ,new State("Pennsylvania" ,"PA")
           ,new State("Rhode Island" ,"RI")
           ,new State("South Carolina","SC")
           ,new State("South Dakota" ,"SD")
           ,new State("Tennessee"  ,"TN")
           ,new State("Texas","TX")
           ,new State("Utah","UT")
           ,new State("Vermont" ,"VT")
           ,new State("Virginia","VA")
           ,new State("Washington" ,"WA")
           ,new State("West Virginia","WV")
           ,new State("Wisconsin"  ,"WI")
           ,new State("Wyoming" ,"WY")
           } ;

        public ComboBoxBinding() {

            // Required by the Windows Forms Designer
            InitializeComponent();

            //Fill the DataSet
            SqlConnection con = new SqlConnection("server=(local)\\NetSDK;Integrated Security=SSPI;database=northwind");
            SqlDataAdapter cmd = new SqlDataAdapter("Select * from Customers where country='USA'", con);
	    try {
                cmd.Fill(customersDataSet1, "Customers");

                //Set up the Combobox bindings
                comboBoxState.DataSource=States;            //Populate the list
                comboBoxState.DisplayMember="LongName";     //Define the field to be displayed
                comboBoxState.ValueMember="ShortName";      //Define the field to be used as the value

                //Bind the selected value of the combobox to the Region field of the current
                //Customer
                comboBoxState.DataBindings.Add("SelectedValue", customersDataSet1, "Customers.Region");

                //Set up the rest of the form bindings
                textBoxID.DataBindings.Add("Text", customersDataSet1, "Customers.CustomerID");
                textBoxCity.DataBindings.Add("Text", customersDataSet1, "Customers.City");
                textBoxTitle.DataBindings.Add("Text", customersDataSet1, "Customers.ContactTitle");
                textBoxAddress.DataBindings.Add("Text", customersDataSet1, "Customers.Address");
                textBoxCompany.DataBindings.Add("Text", customersDataSet1, "Customers.CompanyName");
                textBoxContact.DataBindings.Add("Text", customersDataSet1, "Customers.ContactName");
                textBoxZip.DataBindings.Add("Text", customersDataSet1, "Customers.PostalCode");

                //We want to handle position changing events for the DATA VCR Panel
                this.BindingContext[customersDataSet1,"Customers"].PositionChanged += new System.EventHandler(customers_PositionChanged);

                //Set up the initial text for the DATA VCR Panel
                textBoxPosition.Text = "Record " + (this.BindingContext[customersDataSet1,"Customers"].Position + 1) + " of " + customersDataSet1.Customers.Count;

                //Set the minimum form size
                this.MinimumSize = new Size(375, 361);
	    }
	    catch (Exception ex){
                throw (ex);
	    }
	    finally{
                con.Close();
	    }
        }


        //When the MoveFirst button is clicked set the position for the Customers table
        //to the first record
        private void buttonMoveFirst_Click(object sender, System.EventArgs e) {
            this.BindingContext[customersDataSet1,"Customers"].Position = 0 ;
        }


        //When the MoveLast button is clicked set the position for the Customers table
        //to the last record
        private void buttonMoveLast_Click(object sender, System.EventArgs e) {
            this.BindingContext[customersDataSet1,"Customers"].Position = customersDataSet1.Customers.Count - 1;
        }


        //When the MoveNext button is clicked increment the position for the Customers table
        private void buttonMoveNext_Click(object sender, System.EventArgs e) {
            if (this.BindingContext[customersDataSet1,"Customers"].Position < customersDataSet1.Customers.Count - 1) {
                this.BindingContext[customersDataSet1,"Customers"].Position++;
            }
        }


        //When the MovePrev button is clicked decrement the position for the Customers table
        private void buttonMovePrev_Click(object sender, System.EventArgs e) {
            if (this.BindingContext[customersDataSet1,"Customers"].Position > 0) {
                this.BindingContext[customersDataSet1,"Customers"].Position--;
            }
        }


        //Position has changed - update the DATA VCR panel
        private void customers_PositionChanged(object sender, System.EventArgs e) {
            textBoxPosition.Text = "Record " + (this.BindingContext[customersDataSet1,"Customers"].Position + 1) + " of " + customersDataSet1.Customers.Count;
        }

        protected override void Dispose(bool disposing)
        {
           if (disposing) {
                if (components != null) {
                    components.Dispose();
                }
           }
           base.Dispose(disposing);
        }

        private void InitializeComponent() {
            this.components = new System.ComponentModel.Container();
            this.labelCity = new System.Windows.Forms.Label();
            this.buttonMoveNext = new System.Windows.Forms.Button();
            this.customersDataSet1 = new Microsoft.Samples.WinForms.Cs.ComboBoxBinding.Data.CustomersDataSet();
            this.labelContact = new System.Windows.Forms.Label();
            this.buttonMoveFirst = new System.Windows.Forms.Button();
            this.textBoxID = new System.Windows.Forms.TextBox();
            this.textBoxCity = new System.Windows.Forms.TextBox();
            this.labelContactTitle = new System.Windows.Forms.Label();
            this.labelCompanyName = new System.Windows.Forms.Label();
            this.textBoxTitle = new System.Windows.Forms.TextBox();
            this.textBoxPosition = new System.Windows.Forms.TextBox();
            this.labelState = new System.Windows.Forms.Label();
            this.buttonMovePrev = new System.Windows.Forms.Button();
            this.labelZip = new System.Windows.Forms.Label();
            this.textBoxAddress = new System.Windows.Forms.TextBox();
            this.textBoxCompany = new System.Windows.Forms.TextBox();
            this.panelVCRControl = new System.Windows.Forms.Panel();
            this.comboBoxState = new System.Windows.Forms.ComboBox();
            this.labelAddress = new System.Windows.Forms.Label();
            this.labelID = new System.Windows.Forms.Label();
            this.textBoxContact = new System.Windows.Forms.TextBox();
            this.buttonMoveLast = new System.Windows.Forms.Button();
            this.textBoxZip = new System.Windows.Forms.TextBox();

            this.labelCity.Location = new System.Drawing.Point(16, 176);
            this.labelCity.Size = new System.Drawing.Size(64, 16);
            this.labelCity.TabIndex = 11;
            this.labelCity.Text = "City:";

            this.buttonMoveNext.Anchor = AnchorStyles.Top | AnchorStyles.Right;
            this.buttonMoveNext.Click += new System.EventHandler(buttonMoveNext_Click);
            this.buttonMoveNext.FlatStyle = System.Windows.Forms.FlatStyle.Flat;
            this.buttonMoveNext.Location = new System.Drawing.Point(280, 8);
            this.buttonMoveNext.Size = new System.Drawing.Size(32, 32);
            this.buttonMoveNext.TabIndex = 2;
            this.buttonMoveNext.Text = ">";

            this.customersDataSet1.DataSetName = "CustomersDataSet";

            this.labelContact.Location = new System.Drawing.Point(16, 80);
            this.labelContact.Size = new System.Drawing.Size(64, 16);
            this.labelContact.TabIndex = 7;
            this.labelContact.Text = "Contact:";

            this.buttonMoveFirst.FlatStyle = System.Windows.Forms.FlatStyle.Flat;
            this.buttonMoveFirst.Click += new System.EventHandler(buttonMoveFirst_Click);
            this.buttonMoveFirst.Location = new System.Drawing.Point(8, 8);
            this.buttonMoveFirst.Size = new System.Drawing.Size(32, 32);
            this.buttonMoveFirst.TabIndex = 0;
            this.buttonMoveFirst.Text = "|<";

            this.textBoxID.Anchor = AnchorStyles.Top | AnchorStyles.Left | AnchorStyles.Right;

            this.textBoxID.Enabled = false;
            this.textBoxID.Location = new System.Drawing.Point(88, 16);
            this.textBoxID.ReadOnly = true;
            this.textBoxID.Size = new System.Drawing.Size(299, 20);
            this.textBoxID.TabIndex = 0;
            this.textBoxID.Text = "";

            this.textBoxCity.Anchor = AnchorStyles.Top | AnchorStyles.Left | AnchorStyles.Right;
            this.textBoxCity.Location = new System.Drawing.Point(88, 176);
            this.textBoxCity.Text = "";
            this.textBoxCity.TabIndex = 5;
            this.textBoxCity.Size = new System.Drawing.Size(216, 20);

            this.labelContactTitle.Location = new System.Drawing.Point(16, 112);
            this.labelContactTitle.Size = new System.Drawing.Size(64, 16);
            this.labelContactTitle.TabIndex = 8;
            this.labelContactTitle.Text = "Title:";

            this.labelCompanyName.Location = new System.Drawing.Point(16, 48);
            this.labelCompanyName.Size = new System.Drawing.Size(64, 16);
            this.labelCompanyName.TabIndex = 6;
            this.labelCompanyName.Text = "Company:";

            this.textBoxTitle.Anchor = AnchorStyles.Top | AnchorStyles.Left | AnchorStyles.Right;
            this.textBoxTitle.Location = new System.Drawing.Point(88, 112);
            this.textBoxTitle.Size = new System.Drawing.Size(216, 20);
            this.textBoxTitle.TabIndex = 3;
            this.textBoxTitle.Text = "";

            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.ClientSize = new System.Drawing.Size(464, 357);
            this.Text = "Customer Details";

            this.textBoxPosition.Anchor = AnchorStyles.Left | AnchorStyles.Right;
            this.textBoxPosition.BorderStyle = System.Windows.Forms.BorderStyle.FixedSingle;
            this.textBoxPosition.Enabled = false;
            this.textBoxPosition.Location = new System.Drawing.Point(88, 14);
            this.textBoxPosition.ReadOnly = true;
            this.textBoxPosition.Size = new System.Drawing.Size(184, 20);
            this.textBoxPosition.TabIndex = 1;
            this.textBoxPosition.Text = "";

            this.labelState.Location = new System.Drawing.Point(16, 208);
            this.labelState.Size = new System.Drawing.Size(64, 16);
            this.labelState.TabIndex = 13;
            this.labelState.Text = "State:";

            this.buttonMovePrev.Click += new System.EventHandler(buttonMovePrev_Click);
            this.buttonMovePrev.FlatStyle = System.Windows.Forms.FlatStyle.Flat;
            this.buttonMovePrev.Location = new System.Drawing.Point(48, 8);
            this.buttonMovePrev.Size = new System.Drawing.Size(32, 32);
            this.buttonMovePrev.TabIndex = 1;
            this.buttonMovePrev.Text = "<";

            this.labelZip.Location = new System.Drawing.Point(16, 240);
            this.labelZip.Size = new System.Drawing.Size(64, 16);
            this.labelZip.TabIndex = 12;
            this.labelZip.Text = "Zip:";

            this.textBoxAddress.AcceptsReturn = true;
            this.textBoxAddress.Anchor = AnchorStyles.Top | AnchorStyles.Bottom | AnchorStyles.Left | AnchorStyles.Right;
            this.textBoxAddress.Location = new System.Drawing.Point(88, 144);
            this.textBoxAddress.Size = new System.Drawing.Size(360, 20);
            this.textBoxAddress.Text = "";
            this.textBoxAddress.TabIndex = 4;

            this.textBoxCompany.Anchor = AnchorStyles.Top | AnchorStyles.Left | AnchorStyles.Right;
            this.textBoxCompany.Location = new System.Drawing.Point(88, 48);
            this.textBoxCompany.Size = new System.Drawing.Size(296, 20);
            this.textBoxCompany.Text = "";
            this.textBoxCompany.TabIndex = 1;

            this.panelVCRControl.Anchor = AnchorStyles.Bottom | AnchorStyles.Left | AnchorStyles.Right;
            this.panelVCRControl.AutoScrollMinSize = new System.Drawing.Size(0, 0);
            this.panelVCRControl.BorderStyle = System.Windows.Forms.BorderStyle.FixedSingle;
            this.panelVCRControl.Location = new System.Drawing.Point(88, 288);
            this.panelVCRControl.Size = new System.Drawing.Size(360, 48);
            this.panelVCRControl.TabIndex = 8;
            this.panelVCRControl.Text = "panel1";

            this.comboBoxState.Location = new System.Drawing.Point(88, 208);
            this.comboBoxState.Size = new System.Drawing.Size(176, 20);
            this.comboBoxState.Text = "";
            this.comboBoxState.TabIndex = 6;

            this.labelAddress.Location = new System.Drawing.Point(16, 144);
            this.labelAddress.Size = new System.Drawing.Size(64, 16);
            this.labelAddress.TabIndex = 9;
            this.labelAddress.Text = "Address:";

            this.labelID.Location = new System.Drawing.Point(16, 16);
            this.labelID.Size = new System.Drawing.Size(64, 16);
            this.labelID.TabIndex = 5;
            this.labelID.Text = "ID:";

            this.textBoxContact.Anchor = AnchorStyles.Top | AnchorStyles.Left | AnchorStyles.Right;
            this.textBoxContact.Location = new System.Drawing.Point(88, 80);
            this.textBoxContact.Size = new System.Drawing.Size(339, 20);
            this.textBoxContact.TabIndex = 2;
            this.textBoxContact.Text = "";

            this.buttonMoveLast.Anchor = AnchorStyles.Top | AnchorStyles.Right;
            this.buttonMoveLast.Click += new System.EventHandler(buttonMoveLast_Click);
            this.buttonMoveLast.FlatStyle = System.Windows.Forms.FlatStyle.Flat;
            this.buttonMoveLast.Location = new System.Drawing.Point(320, 8);
            this.buttonMoveLast.Size = new System.Drawing.Size(32, 32);
            this.buttonMoveLast.TabIndex = 3;
            this.buttonMoveLast.Text = ">|";

            this.textBoxZip.Location = new System.Drawing.Point(88, 240);
            this.textBoxZip.Size = new System.Drawing.Size(112, 20);
            this.textBoxZip.TabIndex = 7;
            this.textBoxZip.Text = "";

            this.panelVCRControl.Controls.AddRange(new System.Windows.Forms.Control[] {this.textBoxPosition,
                                                                                          this.buttonMoveFirst,
                                                                                          this.buttonMovePrev,
                                                                                          this.buttonMoveNext,
                                                                                          this.buttonMoveLast});
            this.Controls.AddRange(new System.Windows.Forms.Control[] {this.textBoxZip,
                                                                          this.comboBoxState,
                                                                          this.textBoxCity,
                                                                          this.labelState,
                                                                          this.labelZip,
                                                                          this.labelCity,
                                                                          this.panelVCRControl,
                                                                          this.textBoxCompany,
                                                                          this.labelCompanyName,
                                                                          this.textBoxAddress,
                                                                          this.textBoxTitle,
                                                                          this.textBoxContact,
                                                                          this.textBoxID,
                                                                          this.labelAddress,
                                                                          this.labelContactTitle,
                                                                          this.labelContact,
                                                                          this.labelID});
        }

        [STAThread]
        public static void Main(string[] args) {
            Application.Run(new ComboBoxBinding());
        }

    }
}

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/winforms/samples/data/comboboxbinding/vb/ComboBoxBinding.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------


Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports System.Data
Imports System.Data.SqlClient

Namespace Microsoft.Samples.WinForms.VB.ComboBoxBinding

    Public Class ComboBoxBinding
        Inherits System.Windows.Forms.Form

        ' Define the array of states for the drop-down list
        Public USStates() As USState = New USState() { _
             New USState("Alabama", "AL") _
            , New USState("Alaska", "AK") _
            , New USState("Arizona", "AZ") _
            , New USState("Arkansas", "AR") _
            , New USState("California", "CA") _
            , New USState("Colorado", "CO") _
            , New USState("Connecticut", "CT") _
            , New USState("Delaware", "DE") _
            , New USState("District of Columbia", "DC") _
            , New USState("Florida", "FL") _
            , New USState("Georgia", "GA") _
            , New USState("Hawaii", "HI") _
            , New USState("Idaho", "ID") _
            , New USState("Illinois", "IL") _
            , New USState("Indiana", "IN") _
            , New USState("Iowa", "IA") _
            , New USState("Kansas", "KS") _
            , New USState("Kentucky", "KY") _
            , New USState("Louisiana", "LA") _
            , New USState("Maine", "ME") _
            , New USState("Maryland", "MD") _
            , New USState("Massachusetts", "MA") _
            , New USState("Michigan", "MI") _
            , New USState("Minnesota", "MN") _
            , New USState("Mississippi", "MS") _
            , New USState("Missouri", "MO") _
            , New USState("Montana", "MT") _
            , New USState("Nebraska", "NE") _
            , New USState("Nevada", "NV") _
            , New USState("New Hampshire", "NH") _
            , New USState("New Jersey", "NJ") _
            , New USState("New Mexico", "NM") _
            , New USState("New York", "NY") _
            , New USState("North Carolina", "NC") _
            , New USState("North Dakota", "ND") _
            , New USState("Ohio", "OH") _
            , New USState("Oklahoma", "OK") _
            , New USState("Oregon", "OR") _
            , New USState("Pennsylvania", "PA") _
            , New USState("Rhode Island", "RI") _
            , New USState("South Carolina", "SC") _
            , New USState("South Dakota", "SD") _
            , New USState("Tennessee", "TN") _
            , New USState("Texas", "TX") _
            , New USState("Utah", "UT") _
            , New USState("Vermont", "VT") _
            , New USState("Virginia", "VA") _
            , New USState("Washington", "WA") _
            , New USState("West Virginia", "WV") _
            , New USState("Wisconsin", "WI") _
            , New USState("Wyoming", "WY")}

        Shared Sub Main()
            Application.Run(New ComboBoxBinding())
        End Sub


        Public Sub New()
            MyBase.New()

            ComboBoxBinding = Me

            '  Required by the Windows Forms Designer
            InitializeComponent()

            ' Fill the DataSet
            Dim con As SqlConnection = New SqlConnection("server=(local)\NetSDK;Integrated Security=SSPI;database=northwind")
            Dim cmd As SqlDataAdapter = New SqlDataAdapter("Select * from Customers where country='USA'", con)
		
	    Try
                cmd.Fill(customersDataSet1, "Customers")

                ' Set up the Combobox bindings
                comboBoxUSState.DataSource = USStates            ' Populate the list
                comboBoxUSState.DisplayMember = "LongName"     ' Define the field to be displayed
                comboBoxUSState.ValueMember = "ShortName"      ' Define the field to be used as the value

                ' Bind the selected value of the combobox to the Region field of the current
                ' Customer
                comboBoxUSState.DataBindings.Add("SelectedValue", customersDataSet1, "Customers.Region")

                ' Set up the rest of the form bindings
                textBoxID.DataBindings.Add("Text", customersDataSet1, "Customers.CustomerID")
                textBoxCity.DataBindings.Add("Text", customersDataSet1, "Customers.City")
                textBoxTitle.DataBindings.Add("Text", customersDataSet1, "Customers.ContactTitle")
                textBoxAddress.DataBindings.Add("Text", customersDataSet1, "Customers.Address")
                textBoxCompany.DataBindings.Add("Text", customersDataSet1, "Customers.CompanyName")
                textBoxContact.DataBindings.Add("Text", customersDataSet1, "Customers.ContactName")
                textBoxZip.DataBindings.Add("Text", customersDataSet1, "Customers.PostalCode")

                ' We want to handle position changing events for the DATA VCR Panel
                AddHandler Me.BindingContext(customersDataSet1, "Customers").PositionChanged, AddressOf customers_PositionChanged

                ' Set up the initial text for the DATA VCR Panel
                textBoxPosition.Text = "Record " + (Me.BindingContext(customersDataSet1, "Customers").Position + 1).ToString() + " of " + customersDataSet1.Customers.Count.ToString()

                ' Set the minimum form size
                Me.MinimumSize = New Size(375, 361)
	    Catch ex as Exception
                Throw ex
	    Finally
                con.Close()
	    End Try

        End Sub


        ' When the MoveFirst button is clicked set the position for the Customers table
        ' to the first record
        Protected Sub buttonMoveFirst_Click(ByVal sender As Object, ByVal e As System.EventArgs)
            Me.BindingContext(customersDataSet1, "Customers").Position = 0
        End Sub


        ' When the MoveLast button is clicked set the position for the Customers table
        ' to the last record
        Protected Sub buttonMoveLast_Click(ByVal sender As Object, ByVal e As System.EventArgs)
            Me.BindingContext(customersDataSet1, "Customers").Position = customersDataSet1.Customers.Count - 1
        End Sub


        ' When the MoveNext button is clicked increment the position for the Customers table
        Protected Sub buttonMoveNext_Click(ByVal sender As Object, ByVal e As System.EventArgs)
            If Me.BindingContext(customersDataSet1, "Customers").Position < customersDataSet1.Customers.Count - 1 Then
                Me.BindingContext(customersDataSet1, "Customers").Position += 1
            End If
        End Sub


        ' When the MovePrev button is clicked decrement the position for the Customers table
        Protected Sub buttonMovePrev_Click(ByVal sender As Object, ByVal e As System.EventArgs)
            If Me.BindingContext(customersDataSet1, "Customers").Position > 0 Then
                Me.BindingContext(customersDataSet1, "Customers").Position -= 1
            End If
        End Sub


        ' Position has changed - update the DATA VCR panel
        Protected Sub customers_PositionChanged(ByVal sender As Object, ByVal e As System.EventArgs)
            textBoxPosition.Text = "Record " + (Me.BindingContext(customersDataSet1, "Customers").Position + 1).ToString() + " of " + customersDataSet1.Customers.Count.ToString()
        End Sub

        Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
            If disposing Then
                If Not (components Is Nothing) Then
                    components.Dispose()
                End If
            End If
            MyBase.Dispose(disposing)
        End Sub



#Region " Windows Form Designer generated code "

        Private components As System.ComponentModel.Container
        Private WithEvents customersDataSet1 As Microsoft.Samples.WinForms.VB.ComboBoxBinding.Data.CustomersDataSet
        Private WithEvents textBoxZip As System.Windows.Forms.TextBox
        Private WithEvents comboBoxUSState As System.Windows.Forms.ComboBox
        Private WithEvents textBoxCity As System.Windows.Forms.TextBox
        Private WithEvents labelUSState As System.Windows.Forms.Label
        Private WithEvents labelZip As System.Windows.Forms.Label
        Private WithEvents labelCity As System.Windows.Forms.Label
        Private WithEvents textBoxPosition As System.Windows.Forms.TextBox
        Private WithEvents buttonMoveFirst As System.Windows.Forms.Button
        Private WithEvents buttonMovePrev As System.Windows.Forms.Button
        Private WithEvents buttonMoveNext As System.Windows.Forms.Button
        Private WithEvents buttonMoveLast As System.Windows.Forms.Button
        Private WithEvents textBoxCompany As System.Windows.Forms.TextBox
        Private WithEvents labelCompanyName As System.Windows.Forms.Label
        Private WithEvents textBoxAddress As System.Windows.Forms.TextBox
        Private WithEvents textBoxTitle As System.Windows.Forms.TextBox
        Private WithEvents textBoxContact As System.Windows.Forms.TextBox
        Private WithEvents textBoxID As System.Windows.Forms.TextBox
        Private WithEvents labelAddress As System.Windows.Forms.Label
        Private WithEvents labelContactTitle As System.Windows.Forms.Label
        Private WithEvents labelContact As System.Windows.Forms.Label
        Private WithEvents labelID As System.Windows.Forms.Label
        Private WithEvents panelVCRControl As System.Windows.Forms.Panel

        Private WithEvents ComboBoxBinding As System.Windows.Forms.Form

        Private Sub InitializeComponent()
            Me.components = New System.ComponentModel.Container()
            Me.labelCity = New System.Windows.Forms.Label()
            Me.buttonMoveNext = New System.Windows.Forms.Button()
            Me.customersDataSet1 = New Microsoft.Samples.WinForms.VB.ComboBoxBinding.Data.CustomersDataSet()
            Me.labelContact = New System.Windows.Forms.Label()
            Me.buttonMoveFirst = New System.Windows.Forms.Button()
            Me.textBoxID = New System.Windows.Forms.TextBox()
            Me.textBoxCity = New System.Windows.Forms.TextBox()
            Me.labelContactTitle = New System.Windows.Forms.Label()
            Me.labelCompanyName = New System.Windows.Forms.Label()
            Me.textBoxTitle = New System.Windows.Forms.TextBox()
            Me.textBoxPosition = New System.Windows.Forms.TextBox()
            Me.labelUSState = New System.Windows.Forms.Label()
            Me.buttonMovePrev = New System.Windows.Forms.Button()
            Me.labelZip = New System.Windows.Forms.Label()
            Me.textBoxAddress = New System.Windows.Forms.TextBox()
            Me.textBoxCompany = New System.Windows.Forms.TextBox()
            Me.panelVCRControl = New System.Windows.Forms.Panel()
            Me.comboBoxUSState = New System.Windows.Forms.ComboBox()
            Me.labelAddress = New System.Windows.Forms.Label()
            Me.labelID = New System.Windows.Forms.Label()
            Me.textBoxContact = New System.Windows.Forms.TextBox()
            Me.buttonMoveLast = New System.Windows.Forms.Button()
            Me.textBoxZip = New System.Windows.Forms.TextBox()

            Me.labelCity.Size = New System.Drawing.Size(64, 16)
            Me.labelCity.Location = New System.Drawing.Point(16, 176)
            Me.labelCity.TabIndex = 11
            Me.labelCity.Text = "City:"

            Me.buttonMoveNext.Location = New System.Drawing.Point(280, 8)
            Me.buttonMoveNext.FlatStyle = System.Windows.Forms.FlatStyle.Flat
            Me.buttonMoveNext.Size = New System.Drawing.Size(32, 32)
            Me.buttonMoveNext.TabIndex = 2
            Me.buttonMoveNext.Anchor = AnchorStyles.Top Or AnchorStyles.Right
            Me.buttonMoveNext.Text = ">"
            AddHandler Me.buttonMoveNext.Click, AddressOf Me.buttonMoveNext_Click

            Me.customersDataSet1.DataSetName = "CustomersDataSet"

            Me.labelContact.Size = New System.Drawing.Size(64, 16)
            Me.labelContact.Location = New System.Drawing.Point(16, 80)
            Me.labelContact.TabIndex = 7
            Me.labelContact.Text = "Contact:"

            Me.buttonMoveFirst.Location = New System.Drawing.Point(8, 8)
            Me.buttonMoveFirst.FlatStyle = System.Windows.Forms.FlatStyle.Flat
            Me.buttonMoveFirst.Size = New System.Drawing.Size(32, 32)
            Me.buttonMoveFirst.TabIndex = 0
            Me.buttonMoveFirst.Text = "|<"
            AddHandler Me.buttonMoveFirst.Click, AddressOf Me.buttonMoveFirst_Click

            Me.textBoxID.Text = ""
            Me.textBoxID.Enabled = False
            Me.textBoxID.ReadOnly = True
            Me.textBoxID.TabIndex = 0
            Me.textBoxID.Anchor = AnchorStyles.Top Or AnchorStyles.Left Or AnchorStyles.Right
            Me.textBoxID.Size = New System.Drawing.Size(299, 20)
            Me.textBoxID.Location = New System.Drawing.Point(88, 16)

            Me.textBoxCity.Text = ""
            Me.textBoxCity.TabIndex = 5
            Me.textBoxCity.Size = New System.Drawing.Size(216, 20)
            Me.textBoxCity.Anchor = AnchorStyles.Top Or AnchorStyles.Left Or AnchorStyles.Right
            Me.textBoxCity.Location = New System.Drawing.Point(88, 176)

            Me.labelContactTitle.Size = New System.Drawing.Size(64, 16)
            Me.labelContactTitle.Location = New System.Drawing.Point(16, 112)
            Me.labelContactTitle.TabIndex = 8
            Me.labelContactTitle.Text = "Title:"

            Me.labelCompanyName.Size = New System.Drawing.Size(64, 16)
            Me.labelCompanyName.Location = New System.Drawing.Point(16, 48)
            Me.labelCompanyName.TabIndex = 6
            Me.labelCompanyName.Text = "Company:"

            Me.textBoxTitle.Text = ""
            Me.textBoxTitle.TabIndex = 3
            Me.textBoxTitle.Anchor = AnchorStyles.Top Or AnchorStyles.Left Or AnchorStyles.Right
            Me.textBoxTitle.Size = New System.Drawing.Size(216, 20)
            Me.textBoxTitle.Location = New System.Drawing.Point(88, 112)

            Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
            Me.Text = "Customer Details"
            Me.ClientSize = New System.Drawing.Size(464, 357)

            Me.textBoxPosition.Text = ""
            Me.textBoxPosition.BorderStyle = System.Windows.Forms.BorderStyle.FixedSingle
            Me.textBoxPosition.Enabled = False
            Me.textBoxPosition.ReadOnly = True
            Me.textBoxPosition.TabIndex = 1
            Me.textBoxPosition.Anchor = AnchorStyles.Left Or AnchorStyles.Right
            Me.textBoxPosition.Size = New System.Drawing.Size(184, 20)
            Me.textBoxPosition.Location = New System.Drawing.Point(88, 14)

            Me.labelUSState.Size = New System.Drawing.Size(64, 16)
            Me.labelUSState.Location = New System.Drawing.Point(16, 208)
            Me.labelUSState.TabIndex = 13
            Me.labelUSState.Text = "USState:"

            Me.buttonMovePrev.Location = New System.Drawing.Point(48, 8)
            Me.buttonMovePrev.FlatStyle = System.Windows.Forms.FlatStyle.Flat
            Me.buttonMovePrev.Size = New System.Drawing.Size(32, 32)
            Me.buttonMovePrev.TabIndex = 1
            Me.buttonMovePrev.Text = "<"
            AddHandler Me.buttonMovePrev.click, AddressOf Me.buttonMovePrev_Click

            Me.labelZip.Size = New System.Drawing.Size(64, 16)
            Me.labelZip.Location = New System.Drawing.Point(16, 240)
            Me.labelZip.TabIndex = 12
            Me.labelZip.Text = "Zip:"

            Me.textBoxAddress.Text = ""
            Me.textBoxAddress.AcceptsReturn = True
            Me.textBoxAddress.TabIndex = 4
            Me.textBoxAddress.Anchor = AnchorStyles.Top Or AnchorStyles.Bottom Or AnchorStyles.Left Or AnchorStyles.Right
            Me.textBoxAddress.Size = New System.Drawing.Size(360, 20)
            Me.textBoxAddress.Location = New System.Drawing.Point(88, 144)

            Me.textBoxCompany.Text = ""
            Me.textBoxCompany.TabIndex = 1
            Me.textBoxCompany.Anchor = AnchorStyles.Top Or AnchorStyles.Left Or AnchorStyles.Right
            Me.textBoxCompany.Size = New System.Drawing.Size(296, 20)
            Me.textBoxCompany.Location = New System.Drawing.Point(88, 48)

            Me.panelVCRControl.BorderStyle = System.Windows.Forms.BorderStyle.FixedSingle
            Me.panelVCRControl.Location = New System.Drawing.Point(88, 288)
            Me.panelVCRControl.AutoScrollMinSize = New System.Drawing.Size(0, 0)
            Me.panelVCRControl.Size = New System.Drawing.Size(360, 48)
            Me.panelVCRControl.TabIndex = 8
            Me.panelVCRControl.Anchor = AnchorStyles.Bottom Or AnchorStyles.Left Or AnchorStyles.Right
            Me.panelVCRControl.Text = "panel1"

            Me.comboBoxUSState.Text = ""
            Me.comboBoxUSState.TabIndex = 6
            Me.comboBoxUSState.Size = New System.Drawing.Size(176, 20)
            Me.comboBoxUSState.Location = New System.Drawing.Point(88, 208)

            Me.labelAddress.Size = New System.Drawing.Size(64, 16)
            Me.labelAddress.Location = New System.Drawing.Point(16, 144)
            Me.labelAddress.TabIndex = 9
            Me.labelAddress.Text = "Address:"

            Me.labelID.Size = New System.Drawing.Size(64, 16)
            Me.labelID.Location = New System.Drawing.Point(16, 16)
            Me.labelID.TabIndex = 5
            Me.labelID.Text = "ID:"

            Me.textBoxContact.Text = ""
            Me.textBoxContact.TabIndex = 2
            Me.textBoxContact.Anchor = AnchorStyles.Top Or AnchorStyles.Left Or AnchorStyles.Right
            Me.textBoxContact.Size = New System.Drawing.Size(339, 20)
            Me.textBoxContact.Location = New System.Drawing.Point(88, 80)

            Me.buttonMoveLast.Location = New System.Drawing.Point(320, 8)
            Me.buttonMoveLast.FlatStyle = System.Windows.Forms.FlatStyle.Flat
            Me.buttonMoveLast.Size = New System.Drawing.Size(32, 32)
            Me.buttonMoveLast.TabIndex = 3
            Me.buttonMoveLast.Anchor = AnchorStyles.Top Or AnchorStyles.Right
            Me.buttonMoveLast.Text = ">|"
            AddHandler Me.buttonMoveLast.click, AddressOf Me.buttonMoveLast_Click

            Me.textBoxZip.Text = ""
            Me.textBoxZip.TabIndex = 7
            Me.textBoxZip.Size = New System.Drawing.Size(112, 20)
            Me.textBoxZip.Location = New System.Drawing.Point(88, 240)

            Me.panelVCRControl.Controls.AddRange(New System.Windows.Forms.Control() {textBoxPosition, buttonMovefirst, buttonmoveprev, buttonmovenext, buttonmovelast})
            Me.Controls.AddRange(New System.Windows.Forms.Control() {textBoxZip, comboboxusstate, textboxcity, labelusstate, labelzip, labelcity, panelvcrcontrol, textBoxCompany, labelCompanyName, textBoxAddress, textBoxTitle, textBoxContact, textBoxID, labelAddress, labelContactTitle, labelContact, labelID})
        End Sub

#End Region

    End Class

End Namespace

















